subprojects/dependency-management/src/main/java/org/gradle/internal/component/AmbiguousVariantSelectionException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.component;
+
+/**
+ * This type is {@code deprecated} and will be removed in Gradle 9.0.
+ *
+ * This is temporarily available for migration only.  Please use {@link AmbiguousArtifactVariantsException} instead.
```

Comment:
```suggestion
 * This is temporarily available for migration only.
 * This exception class is internal and has been replaced by {@link AmbiguousArtifactVariantsException}, which is also internal. 
 * If possible, catch a {@link RuntimeException} instead to avoid depending on Gradle internal classes.
```

Thoughts?

I worry that recommending users to reference the new internal class can lead to further issues like this in the future. 

I think users will do whatever they want, including depending on internal classes. However, if we can get them to only catch `RuntimeException` and that is acceptable for them, we should try to push for that resolution.

Reply:
Good advice.